docs/architecture.md

```diff
@@ -0,0 +1,96 @@
+# Pulp CLI Architecture
+
+## Plugin system
+
+The Pulp CLI is designed with a plugin structure. Plugins can either live in the main package or be shipped independently.
```

Comment:
```suggestion
The Pulp CLI is designed with a plugin structure. Plugins can either live in the pulp-cli package or be shipped independently.
```

Reply:
I'm assuming that `main` is the module name and pulp-cli is the name of the package. Or maybe I misunderstand?

Reply:
`main` is only the command entrypoint to be precise. And precise is good!